packages/react-dom/src/client/ReactDOMRoot.js

```diff
@@ -190,6 +192,18 @@ export function createRoot(
   return new ReactDOMRoot(root);
 }
 
+function ReactDOMHydrationRoot(internalRoot: FiberRoot) {
+  ReactDOMRoot.call(this, internalRoot);
+}
+function scheduleHydration(target: Node) {
+  if (target) {
+    queueExplicitHydrationTarget(target);
+  }
+}
+Object.assign(ReactDOMHydrationRoot.prototype, ReactDOMRoot.prototype, {
```

Comment:
Nit: Instead of `Object.assign` let's assign each method like this:

```js
ReactDOMRoot.prototype.render = ReactDOMHydrationRoot.prototype.render = function render() {...}
```

I feel like this is slightly easier to tell which methods are available on each root, instead of modeling it as a subclass. Not sure it always will be a subclass, like maybe we will add methods to client-only roots that don't exist on a hydration root.